gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClientV1Impl.java

```diff
@@ -239,8 +240,13 @@ public KubernetesClientV1Impl(Vertx vertx, KubernetesConfig kubernetesConfig) {
         Watch watch = new Watch();
         watchMap.putIfAbsent(watch.uid, watch);
 
+        String flowableKey = fieldSelector.equals("") ? resource.namespace : resource.name;
+        if (flowableMap.containsKey(flowableKey)) {
+            return (Flowable<T>) flowableMap.get(flowableKey);
+        }
+
         LOGGER.info("Start watching namespace [{}] with fieldSelector [{}]", resource.namespace, fieldSelector);
-        return Flowable
+        Flowable<T> flowable = Flowable
             .<T>create(emitter -> fetchEvents(emitter, resource, fieldSelector, watch.uid, type), BackpressureStrategy.BUFFER)
```

Comment:
@jhaeyaert Shall I change the BackpressureStrategy to "LATEST" because this is actually what we are interested in in here in the case of a wacher? The most recent values in Kubernetes and we don't care about the old one at all

Reply:
Yeah it makes sense indeed 👍